src/benchmarks/challenges.ts

```diff
@@ -0,0 +1,213 @@
+        text: `
+// TODO: Implement the multiply function
+export function multiply(a, b) {
+  // TODO
+}
+      `,
+      },
+    },
+  },
+  {
+    name: "square",
+    prompt: "Implement the 'square' function that returns x*x.",
+    files: {
+      file1: {
+        name: "index.mjs",
+        text: `
+import { square } from "./functions.mjs";
+
+const input = 5;
+const result = square(input);
+if (result !== 25) {
+  console.error("Test failed: expected 25, got", result);
+  process.exit(1);
+}
+console.log("Square test passed");
+process.exit(0);
+      `,
+      },
+      file2: {
+        name: "functions.mjs",
+        text: `
+// TODO: Implement the square function
+export function square(x) {
+  // TODO
+}
+      `,
+      },
+    },
+  },
+  {
+    name: "toUpperCase",
+    prompt:
+      "Implement the toUpperCase function that returns the given string in uppercase.",
+    files: {
+      file1: {
+        name: "index.mjs",
+        text: `
+import { toUpperCase } from "./functions.mjs";
+
+const input = "hello";
+const result = toUpperCase(input);
+if (result !== "HELLO") {
+  console.error("Test failed: expected 'HELLO', got", result);
+  process.exit(1);
+}
+console.log("toUpperCase test passed");
+process.exit(0);
+      `,
+      },
+      file2: {
+        name: "functions.mjs",
+        text: `
+// TODO: Implement the toUpperCase function
+export function toUpperCase(str) {
+  // TODO
+}
+      `,
+      },
+    },
+  },
+  {
+    name: "reverseString",
+    prompt:
+      "Implement the reverseString function that reverses the given string.",
+    files: {
+      file1: {
+        name: "index.mjs",
+        text: `
+import { reverseString } from "./functions.mjs";
+
+const input = "OpenAI";
+const expected = "IAnepO";
+
+const result = reverseString(input);
+if (result !== expected) {
+  console.error("Test failed: expected", expected, "but got", result);
+  process.exit(1);
+}
+console.log("reverseString test passed");
+process.exit(0);
+      `,
+      },
+      file2: {
+        name: "functions.mjs",
+        text: `
+// TODO: Implement the reverseString function
+export function reverseString(str) {
+  // TODO
+}
+      `,
+      },
+    },
+  },
+  {
+    name: "stockPriceChart",
+    type: "visualization",
+    prompt:
+      "Create a bar chart showing AAPL stock prices over time. The chart should:\n" +
+      "1. Use D3.js for visualization\n" +
+      "2. Have proper axes with labels\n" +
+      "3. Use a pleasing color scheme\n" +
+      "4. Include a title\n" +
+      "5. Show price values on top of each bar\n" +
+      "Parse the CSV data from the 'data.csv' file and create an HTML file that renders the visualization.",
+    sampleData: stockData,
+    files: {
+      file1: {
+        name: "index.html",
+        text: `<!DOCTYPE html>
+<html>
+<head>
+  <title>Stock Price Visualization</title>
+  <script src="https://d3js.org/d3.v7.min.js"></script>
+  <!-- TODO: Add your styles and chart implementation -->
```

Comment:
Love the vanilla minimal setup! The AI seems to be really good with this starter, and less good with more complex starters, which is something I'd love to quantify - how much better.